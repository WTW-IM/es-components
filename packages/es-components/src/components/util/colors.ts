import tinycolor, { ColorInput } from 'tinycolor2';

export function darken(color: ColorInput, percent: number) {
  return tinycolor(color).darken(percent).toHexString();
}

export function lighten(color: ColorInput, percent: number) {
  return tinycolor(color).lighten(percent).toHexString();
}

export function getTextColor(
  bgColor: ColorInput,
  darkestColor: ColorInput = 'black',
  lightestColor: ColorInput = 'white'
) {
  return tinycolor.readability(darkestColor, bgColor) > 10
    ? darkestColor
    : lightestColor;
}
