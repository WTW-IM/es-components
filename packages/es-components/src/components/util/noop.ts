export default function noop() {
  // noop
}
