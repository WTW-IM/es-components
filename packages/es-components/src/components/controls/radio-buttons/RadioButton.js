import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Label from '../label/Label';
import getRadioFillVariables from './radio-fill-variables';
import { useTheme } from '../../util/useTheme';
import ValidationContext from '../ValidationContext';
import useUniqueId from '../../util/useUniqueId';

function radioFill(color) {
  return `
    background-color: ${color};
    border-radius: 50%;
    content: '';
    display: block;
    height: 13px;
    width: 13px;
    transition: background 0.25s linear;
  `;
}

const RadioLabel = styled(Label)`
  align-self: flex-start;
  cursor: pointer;
  display: flex;
  font-family: 'Source Sans Pro', 'Segoe UI', Segoe, Calibri, Tahoma, sans-serif;
  font-size: ${props => props.theme.font.baseFontSize};
  font-weight: normal;
  line-height: ${props => props.theme.font.baseLineHeight};
  margin-right: 15px;
  margin-bottom: 10px;
  position: relative;
  padding: 10px 0 10px 10px;
  text-transform: none;

  &:hover > .es-radio__fill:before {
    ${props => !props.checked && radioFill(props.hoverFillColor)};
  }

  @media (min-width: ${props => props.theme.screenSize.tablet}) {
    padding: 5px 0;
  }
`;

const RadioInput = styled.input`
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
  position: absolute;

  &:focus ~ .es-radio__fill {
    box-shadow: 0 0 3px 3px ${props => props.theme.colors.inputFocus};
  }
`;

const RadioDisplay = styled.span`
  align-items: center;
  border: 3px solid ${props => props.borderColor};
  border-radius: 50%;
  box-sizing: border-box;
  display: flex;
  height: 25px;
  justify-content: center;
  margin-right: 8px;
  min-width: 25px;

  &:before {
    ${props => radioFill(props.fill)};
  }
`;

export function RadioButton({ children, className, ...radioProps }) {
  const id = useUniqueId(radioProps.id);
  const isChecked = radioProps.checked;
  const theme = useTheme();
  const validationState = React.useContext(ValidationContext);
  const { hover, fill } = getRadioFillVariables(
    isChecked,
    radioProps.disabled,
    validationState,
    theme.colors
  );
  const radioDisplayFill = isChecked ? fill : theme.colors.white;

  const labelProps = {
    disabled: radioProps.disabled,
    htmlFor: id,
    hoverFillColor: hover,
    validationState,
    checked: isChecked
  };

  const inputProps = {
    ...radioProps,
    checked: isChecked
  };

  return (
    <RadioLabel className={className} {...labelProps}>
      <RadioInput type="radio" id={id} {...inputProps} />
      <RadioDisplay
        className="es-radio__fill"
        borderColor={fill}
        fill={radioDisplayFill}
      />
      {children}
    </RadioLabel>
  );
}

export const propTypes = {
  ...(RadioInput.propTypes || {}),
  /** supports styled-component usage, applies to the wrapping Label */
  className: PropTypes.string,
  children: PropTypes.node
};

RadioButton.propTypes = propTypes;

export default RadioButton;
