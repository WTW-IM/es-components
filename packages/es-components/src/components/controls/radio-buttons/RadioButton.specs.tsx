import React from 'react';
import * as CSS from 'csstype';
import { screen } from '@testing-library/react';
// using a different name for `css` so prettier doesn't mess with css selectors
import { css as cssStyled } from 'styled-components';
import { renderWithTheme } from '../../util/test-utils';
import RadioButton, {
  RadioDisplay,
  RadioDisplayWrapper,
  RadioInput
} from './RadioButton';
import ValidationContext from '../ValidationContext';
import { ValidationStyleType } from 'es-components-shared-types';
import viaTheme from 'es-components-via-theme';
import noop from '../../util/noop';
import { lighten } from '../../util/colors';

describe('RadioButton colors', () => {
  interface ColorTest {
    disabled: boolean;
    checked: boolean;
    validationState: ValidationStyleType;
    expectedColor: CSS.Property.BorderColor;
  }

  test.each<ColorTest>([
    {
      disabled: false,
      checked: false,
      validationState: 'default',
      expectedColor: viaTheme.validationTextColor.default
    },
    {
      disabled: true,
      checked: false,
      validationState: 'default',
      expectedColor: viaTheme.validationTextColor.default
    },
    {
      disabled: false,
      checked: true,
      validationState: 'default',
      expectedColor: viaTheme.colors.primary
    },
    {
      disabled: true,
      checked: true,
      validationState: 'default',
      expectedColor: viaTheme.colors.primary
    },
    {
      disabled: false,
      checked: false,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: true,
      checked: false,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: false,
      checked: true,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    },
    {
      disabled: true,
      checked: true,
      validationState: 'warning',
      expectedColor: viaTheme.validationTextColor.warning
    }
  ])(
    `renders correct colors for { disabled: $disabled, checked: $checked, validationState: '$validationState' }`,
    async ({ disabled, checked, validationState, expectedColor }) => {
      const disabledColor = viaTheme.colors.gray5;
      const expectedHoverColor = lighten(expectedColor, 40);
      renderWithTheme(
        <ValidationContext.Provider value={validationState}>
          <RadioButton
            {...{ disabled, checked, onChange: noop, id: 'test-radio' }}
          >
            Option
          </RadioButton>
        </ValidationContext.Provider>
      );

      const radio = await screen.findByRole('radio');
      const label = await screen.findByText('Option', {
        selector: 'label'
      });

      // Spacing is important in the below `modifier` properties.
      // In failing scenarios, use the snapshots to verify the correct string with spaces is being used.
      expect(radio).toMatchSnapshot();
      expect(label).toMatchSnapshot();

      expect(radio).toHaveStyleRule('border-color', expectedColor, {
        modifier: cssStyled`& ~${RadioDisplay}`
      });

      expect(radio).toHaveStyleRule('background-color', expectedColor, {
        modifier: cssStyled`&:checked~${RadioDisplay}::before`
      });

      expect(radio).toHaveStyleRule('border-color', disabledColor, {
        modifier: cssStyled`&&:disabled ~${RadioDisplay}`
      });

      expect(radio).toHaveStyleRule('background-color', disabledColor, {
        modifier: cssStyled`&&:disabled:checked~${RadioDisplay}::before`
      });

      expect(label).toHaveStyleRule('background-color', expectedHoverColor, {
        modifier: cssStyled`&:hover>${RadioDisplayWrapper}>${RadioInput}:not(:disabled):not(:checked)~${RadioDisplay}::before`
      });
    }
  );
});
