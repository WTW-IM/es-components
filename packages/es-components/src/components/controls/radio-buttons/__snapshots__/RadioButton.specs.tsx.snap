// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'default' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #444;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #444;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #aaaaaa;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  class="c0 c1"
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'default' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #444;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #444;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #aaaaaa;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
  hover="#aaaaaa"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'warning' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #c25400;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #c25400;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  class="c0 c1"
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: false, validationState: 'warning' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #c25400;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #c25400;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
  hover="#ffbf8f"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'default' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #0073b6;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #0073b6;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #83d1ff;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  checked=""
  class="c0 c1"
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'default' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #0073b6;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #0073b6;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #83d1ff;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
  hover="#83d1ff"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'warning' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #c25400;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #c25400;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  checked=""
  class="c0 c1"
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: false, checked: true, validationState: 'warning' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #c25400;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #c25400;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  for="test-radio"
  hover="#ffbf8f"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'default' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #444;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #444;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #aaaaaa;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  class="c0 c1"
  disabled=""
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'default' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #444;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #444;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #aaaaaa;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
  hover="#aaaaaa"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      disabled=""
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'warning' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #c25400;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #c25400;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  class="c0 c1"
  disabled=""
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: false, validationState: 'warning' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #c25400;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #c25400;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
  hover="#ffbf8f"
>
  <div
    class=""
  >
    <input
      class="c2 c3"
      disabled=""
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'default' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #0073b6;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #0073b6;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #83d1ff;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  checked=""
  class="c0 c1"
  disabled=""
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'default' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #0073b6;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #0073b6;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #83d1ff;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
  hover="#83d1ff"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      disabled=""
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'warning' } 1`] = `
.c1 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c1 ~.c3 {
  border-color: #c25400;
}

.c1:focus~.c3 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c1:checked~.c3::before {
  background-color: #c25400;
}

.c1.c1:disabled ~.c3 {
  border-color: #c5c5c5;
}

.c1.c1:disabled:checked~.c3::before {
  background-color: #c5c5c5;
}

.c2:hover>.sc-gjcoXW>.c0:not(:disabled):not(:checked)~.c3::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {

}

@media (min-width: 768px) {

}

<input
  checked=""
  class="c0 c1"
  disabled=""
  id="test-radio"
  name=""
  type="radio"
/>
`;

exports[`RadioButton colors renders correct colors for { disabled: true, checked: true, validationState: 'warning' } 2`] = `
.c0 {
  color: #444;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: bold;
  line-height: 1.428;
  margin-bottom: 5px;
  display: inline-block;
}

.c0[disabled] {
  cursor: not-allowed;
}

.c5 {
  display: flex;
  min-width: 25px;
  height: 25px;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  margin-right: 8px;
}

.c5::before {
  display: block;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: transparent;
  content: '';
  transition: background 0.25s linear;
}

.c3 {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}

.c3 ~.c4 {
  border-color: #c25400;
}

.c3:focus~.c4 {
  box-shadow: 0 0 3px 3px #83bffc;
}

.c3:checked~.c4::before {
  background-color: #c25400;
}

.c3.c3:disabled ~.c4 {
  border-color: #c5c5c5;
}

.c3.c3:disabled:checked~.c4::before {
  background-color: #c5c5c5;
}

.c1 {
  position: relative;
  display: flex;
  align-self: flex-start;
  padding: 10px 0 10px 10px;
  margin-right: 15px;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 1.125rem;
  font-weight: normal;
  line-height: 1.428;
  text-transform: none;
}

.c1:hover>.>.c2:not(:disabled):not(:checked)~.c4::before {
  background-color: #ffbf8f;
}

@media (min-width: 768px) {
  .c0 {
    margin-right: 20px;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 5px 0;
  }
}

<label
  class="c0 c1"
  disabled=""
  for="test-radio"
  hover="#ffbf8f"
>
  <div
    class=""
  >
    <input
      checked=""
      class="c2 c3"
      disabled=""
      id="test-radio"
      name=""
      type="radio"
    />
    <span
      class="c4 c5 es-radio__fill"
    />
  </div>
  Option
</label>
`;
